.github/workflows/test.yml

```diff
@@ -43,3 +44,95 @@ jobs:
           ATLAS_PUBLIC_KEY: ${{ secrets.ATLAS_PUBLIC_KEY }}
           ATLAS_PRIVATE_KEY: ${{ secrets.ATLAS_PRIVATE_KEY }}
 
+  prepare-e2e:
+    name: Prepare E2E configuration and image
+    needs: [unit-test, int-test]
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2.3.1
+
+    - name: Prepare tag
+      id: prepare
+      uses: ./.github/actions/set-tag
+
+    - name: Push Atlas Operator to Registry
+      uses: docker/build-push-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        repository: ${{ secrets.DOCKER_REPO }}
+        registry: ${{ secrets.DOCKER_REGISTRY }}
+        tags: ${{ steps.prepare.outputs.tag }}
+
+  e2e:
+    name: E2E tests
+    needs: prepare-e2e
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # k8s: ["1.17-kind", "1.19-kind", "1.17-opeshift"] # <supported platform version>-<platform>
+        k8s: ["v1.18.15-kind"] # <K8sGitVersion>-<Platform>
+        test: ["all-in-one"] # TODO refactor
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2.3.1
+
+    - name: Prepare tag
+      id: prepare
+      uses: ./.github/actions/set-tag
+
+    - name: Generate configuration for the tests
+      uses: ./.github/actions/gen-install-scripts
+      with:
+        IMAGE_URL: ${{ secrets.DOCKER_REPO }}:${{ steps.prepare.outputs.tag }}
+        ENV: dev
+
+    - name: Set properties
+      id: properties
+      run: |
+        version=$(echo ${{ matrix.k8s }} | awk -F "-" '{print $1}')
+        platform=$(echo ${{ matrix.k8s }} | awk -F "-" '{print $2}')
+        echo "::set-output name=k8s_version::$version"
+        echo "::set-output name=k8s_platform::$platform"
+
+    # run if platform = kind #TODO
+    - name: Create k8s Kind Cluster
```

Comment:
(for discussion)
If we setup the environment in the workflow itself - this will mean that we'll have to duplicate this workflow code for each separate Ginkgo test case that we run.
While if we move the "configure cluster" code into the ginkgo test itself (to the "prepare suite") - this would allow us to have one single workflow job and have all the parallelization inside the ginkgo instead

(let's discuss this with the team)

Reply:
forget that - I think I forgot that we wanted to move the "e2e suite name" to the matrix params. This solves the duplication problem